diff --git a/core/modules/content_translation/src/ContentTranslationHandler.php b/core/modules/content_translation/src/ContentTranslationHandler.php
index 7e6a7ca..8cfe64f 100644
--- a/core/modules/content_translation/src/ContentTranslationHandler.php
+++ b/core/modules/content_translation/src/ContentTranslationHandler.php
@@ -17,6 +17,7 @@
 use Drupal\Core\Session\AccountInterface;
 use Drupal\user\Entity\User;
 use Drupal\user\EntityOwnerInterface;
+use Drupal\Component\Render\FormattableMarkup;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
 /**
@@ -562,7 +563,7 @@ protected function addTranslatabilityClue(&$element) {
     // check is because some elements have a #title attribute even though it is
     // not rendered; for instance, field containers.
     if (isset($element['#type']) && isset($fapi_title_elements[$element['#type']]) && isset($element['#title'])) {
-      $element['#title'] .= $suffix;
+      $element['#title'] = new FormattableMarkup($element['#title'] . $suffix, []);
     }
     // If the current element does not have a (valid) title, try child elements.
     elseif ($children = Element::children($element)) {
@@ -573,7 +574,7 @@ protected function addTranslatabilityClue(&$element) {
     // If there are no children, fall back to the current #title attribute if it
     // exists.
     elseif (isset($element['#title'])) {
-      $element['#title'] .= $suffix;
+      $element['#title'] = new FormattableMarkup($element['#title'] . $suffix, []);
     }
   }
 
